### PR TITLE
Handle REST client exception for unsupported URL

### PR DESCRIPTION
**Pull request by wwelling**
Handles org.springframework.web.client.HttpClientErrorException: 404 when fedora metadata for filename contain special characters.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: src/main/java/edu/tamu/iiif/service/AbstractManifestService.java [hunk 1]
```diff
@@ -165,10 +165,22 @@ protected Model getRdfModel(String url) throws NotFoundException {
     }
 
     protected String getRdf(String url) throws NotFoundException {
-        Optional<String> rdf = Optional.ofNullable(restTemplate.getForObject(url, String.class));
-        if (rdf.isPresent()) {
-            logger.debug("RDF for {}: \n{}\n", url, rdf.get());
-            return rdf.get();
+        try {
+            if (logger.isDebugEnabled()) {
+                logger.info("Requesting RDF for {}", url);
```
Comment (kaladay) [inline src/main/java/edu/tamu/iiif/service/AbstractManifestService.java:170]: Is `logger.debug()` somehow not working?
I would think `logger.debug()` only runs when debug is enabled, helping us avoid the need to explicitly check `logger.isDebugEnabled()`.

File: src/main/java/edu/tamu/iiif/service/AbstractManifestService.java [hunk 2]
```diff
@@ -165,10 +165,22 @@ protected Model getRdfModel(String url) throws NotFoundException {
...
+            }
+            Optional<String> rdf = Optional.ofNullable(restTemplate.getForObject(url, String.class));
+            if (rdf.isPresent()) {
+                if (logger.isDebugEnabled()) {
```
Comment (kaladay) [inline src/main/java/edu/tamu/iiif/service/AbstractManifestService.java:174]: I would think this `logger.isDebugEnabled()` is not necessary when `logger.debug()` is called. Is this not the case?

File: src/main/java/edu/tamu/iiif/service/AbstractManifestService.java [hunk 3]
```diff
@@ -165,10 +165,22 @@ protected Model getRdfModel(String url) throws NotFoundException {
...
+                    logger.debug("RDF for {}: \n{}\n", url, rdf.get());
+                }
+                return rdf.get();
+            }
+        } catch (RestClientException e) {
+            logger.error("Failed to get RDF for {}: {}", url, e.getMessage());
+            if (logger.isDebugEnabled()) {
```
Comment (kaladay) [inline src/main/java/edu/tamu/iiif/service/AbstractManifestService.java:181]: Same situation regarding `logger.isDebugEnabled()` as the comment about on line 174.

File: src/main/java/edu/tamu/iiif/service/AbstractManifestService.java (hunk 1 above)
Comment (wwelling) [reply src/main/java/edu/tamu/iiif/service/AbstractManifestService.java:170]: In this case I only want an info log when debug is enabled.

File: src/main/java/edu/tamu/iiif/service/AbstractManifestService.java (hunk 2 above)
Comment (wwelling) [reply src/main/java/edu/tamu/iiif/service/AbstractManifestService.java:174]: I do not have a definitive answer, but I have switched between using isDebugEnabled as thinking the call to debug is not necessary and incur more cost than the isDebugEnabled call and condition.

**Review by kaladay — APPROVED** [review]

**Comment by wwelling** [comment]
@jcreel seeking approval for release candidate deployment to debug production 500 Internal Server Error a bit further.

**Pull request merged**